### DISCUSSION TITLE
Comparison to FuseSOC

### DISCUSSION THREAD

**Discussion by krupan** [siliconcompiler/siliconcompiler#3075] [Development]
Has anyone compared siliconcompiler with [FuseSoC](https://fusesoc.readthedocs.io/en/stable/index.html#)? I'm curious what the differences and similarities are.

Or is it better to compare siliconcompiler to [Edalize](https://github.com/olofk/edalize) (a subcompnent of FusSoC)?

**Comment by Ciprian167** [comment]
SiliconCompiler is a much more complete solution to the problem FuseSoC is trying to solve. FuseSoC's approach can be constricting at times, mainly because it uses static configuration files (*.core) to capture file lists and arguments. It also lacks many things out of the box, like a test runner and so on. SiliconCompiler embraces Python, and instead of having complicated YAML files to capture cores, everything will be captured using Python. It also supports very complex flows which can be described in a flowgraph.
